File: backend/src/main/java/com/example/backend/community/domain/Community.java
package com.example.backend.community.domain;

import lombok.*;
import org.springframework.data.relational.core.mapping.Table;

import javax.persistence.*;
import java.sql.Timestamp;

@Getter
@Builder
@AllArgsConstructor
@NoArgsConstructor
@Table(name = "TEST_DB")
@Entity
public class Community {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long communityId;

    @Column(nullable = false)
    private String communityWriter;

    @Column(length = 200, nullable = false)
    private String communityImg;

    @Column(nullable = false)
    private String communityContent;

    @Column(nullable = false)
    private Timestamp communityDate;

    @Column(nullable = false)
    private Integer communityLike;

    @Column(nullable = false)
    private String latitude;

    @Column(nullable = false)
    private String longitude;

    public void increaseLike() { this.communityLike++; }

    public void decreaseLike() {
        this.communityLike--;
    }
}

File: frontend/frontend/Models/Community.swift
//
//  Community.swift
//  frontend
//
//  Created by 이정민 on 2023/06/06.
//

import Foundation

struct Community: Hashable, Codable {
  let communityId: Int
  let communityWriter: String
  let communityImg: String
  let communityContent: String
  let communityDate: String
  let communityLike: Int
  let latitude: String
  let longitude: String
}



let communityes: Community = Community(
  communityId : 1234,
  communityWriter : "Google_1",
  communityImg : "image.jpg",
  communityContent : "This is the content of my post.",
  communityDate : "2023-06-23",
  communityLike : 32,
  latitude : "35.888",
  longitude : "128.6136"
)
